flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManagerGateway.java

```diff
@@ -122,4 +122,11 @@ void notifySlotAvailable(
 	 * @param optionalDiagnostics
 	 */
 	void shutDownCluster(final ApplicationStatus finalStatus, final String optionalDiagnostics);
+
+	/**
+	 * sends the heartbeat to resource manager from task manager
+	 * @param resourceID unique id of the task manager
+	 * @param payload the payload information of the task manager
+	 */
+	void sendHeartbeatFromTaskManager(final ResourceID resourceID, final Object payload);
```

Comment:
Better to call this method `reportHeartbeatFromTaskManager` or `heartbeatFromTaskManager`. Maybe we could also introduce a dedicated interface for this which is inherited by the `ResourceManagerGateway`?